internal/upload/koji/koji.go

```diff
@@ -420,3 +424,24 @@ func GSSAPICredentialsFromEnv() (*GSSAPICredentials, error) {
 		KeyTab:    keyTab,
 	}, nil
 }
+
+func CreateKojiTransport(relaxTimeout uint) *http.Transport {
+	// Koji for some reason needs TLS renegotiation enabled.
+	// Clone the default http transport and enable renegotiation.
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.TLSClientConfig = &tls.Config{
+		Renegotiation: tls.RenegotiateOnceAsClient,
+		MinVersion:    tls.VersionTLS12,
+	}
+
+	// Relax timeouts a bit
+	if relaxTimeout > 0 {
+		transport.TLSHandshakeTimeout *= time.Duration(relaxTimeout)
+		transport.DialContext = (&net.Dialer{
+			Timeout:   30 * time.Second * time.Duration(relaxTimeout),
+			KeepAlive: 30 * time.Second,
```

Comment:
Is there a reason to send the keep-alive probes every 30 seconds instead of every 15 seconds (default value)?

Reply:
I thought that's the default https://go.dev/src/net/http/transport.go#L47
mmm making it 15 would increase the traffic but we can try.

Reply:
Ah, I missed this. Makes sense to me to leave this one untouched, unless it gives us problems?

Reply:
Ahhh, such a mess. Yeah, let's leave it untouched and that means 30 seconds.

I accidentally read this documentation (https://pkg.go.dev/net#Dialer) which mentions that the default value for `net.Dialer` is 15 seconds but `DefaultTransport` actually sets it to 30 seconds because one default value is apparently not enough. :shrug: